spec/ParseUser.spec.js

```diff
@@ -1246,6 +1246,20 @@ describe('Parse.User testing', () => {
     done();
   });
 
+  it('log in with provider despite invalid authData field in DB', async done => {
+    const provider = getMockFacebookProvider();
+    Parse.User._registerAuthenticationProvider(provider);
+    const user = await Parse.User._logInWith('facebook');
+    user.set('authData', null);
```

Comment:
Does unset('authData') work here?

Reply:
It works in the sense that when unsetting, the test fails with

> ParseError: 252 This authentication method is unsupported.

I'd have to dig deeper, but unsetting of `authData` should not be supported just like setting of `authData`. The correct ways are `link` and `unlink`.

This PR is merely intended to prevent a server crash if there is an `authData` field in the DB for whatever reason. I'd suggest that in a separate PR we can:

a) evaluate why we don't just use `authData` as a non-synthesized field in the DB to store all provider data and potentially move away from this `_auth_data_*` concept
b) reject direct setting and unsetting of `authData`

Reply:
a) I don’t know why it’s like that. I assume it was to _auth_data into columns in other databases like Postgres that didn’t support json at the time or its been that way from the beginning. We will never know.
b) You can move the logic to RestWrite or UserRouter

Reply:
ad b), the overwriting of the synthesized `authData` with the incorrect `authData` field in the DB happens in the storage adapters, so I'm not sure how to centralize that easily at this point. Do you think we can leave it there for now?